src/algebra/algebra/basic.lean

```diff
@@ -851,6 +851,33 @@ ext $ λ x, show e₁.to_linear_map x = e₂.to_linear_map x, by rw H
 @[simp] lemma trans_to_linear_map (f : A₁ ≃ₐ[R] A₂) (g : A₂ ≃ₐ[R] A₃) :
   (f.trans g).to_linear_map = g.to_linear_map.comp f.to_linear_map := rfl
 
+section of_linear_equiv
+
+variables (l : A₁ ≃ₗ[R] A₂)
+  (map_mul : ∀ x y : A₁, l (x * y) = l x * l y)
+  (commutes : ∀ r : R, l (algebra_map R A₁ r) = algebra_map R A₂ r)
+
+def of_linear_equiv : A₁ ≃ₐ[R] A₂ :=
```

Comment:
Do we have a similar API for `ring_equiv.of_add_equiv` or `linear_equiv.of_add_equiv`?

Reply:
No, we don't. Those are more of the `old_structure`  and `extends` form, so there, a simple `{ ..e }` should work.

Reply:
Then shouldn't we do the same thing here?

Reply:
(and add simp lemmas about `\u{ to_fun := f, inv_fun := f.symm, arbitrary_hyps...}`)

Reply:
Is there a way to make this happen during the `extends` invocation?

Reply:
There's a lot of boilerplate lemmas it would be nice to get automatically, but we're not there yet.

Reply:
> No, we don't. Those are more of the `old_structure` and `extends` form, so there, a simple `{ ..e }` should work.

We do actually, `alg_hom.mk'` is the one from `ring_hom` to `alg_hom`. I suspect we might have a lot of similar primed lemma names.